src/config.py

```diff
@@ -300,12 +300,15 @@ def internal_listener(self) -> Listener:
         return Listener(host=self.charm.unit_host, protocol=protocol, scope="INTERNAL")
 
     @property
-    def extra_listeners(self) -> List[Listener]:
+    def extra_listeners(self) -> List[Listener | None]:
```

Comment:
If we return empty list `[]` the type is still `List[Listener]`. Agreed on not having lists with `None`.

**suggestion:** To avoid nesting, I would suggest restructuring this method as:

```python
if not self.charm.model.relations.get(REL_NAME, None):
    return []

return [
    Listener(host=self.charm.unit_host, protocol=auth, scope="EXTERNAL")
]

```